### PR TITLE
Alanocallaghan master

### PR DESCRIPTION
**Pull request by zlskidmore**
Continuation of #282

### CODE REVIEW

**Comment by zlskidmore** [comment]
@Alanocallaghan Thanks for the work on this, looks great! I made a few minor changes:
1. added vignette documentation for new features
2. fixed issue with x label object missing in new proportion function in specific cases
3. fixed issue with x label title being displayed twice and or missing

please look this over, if you agree with all my changes i'll go ahead and merge and push to the bioconductor devel site.

**Comment by Alanocallaghan** [comment]
Thanks I'll have a look tomorrow evening

File: R/waterfall.R [hunk 1]
```diff
@@ -273,7 +273,8 @@ waterfall <- function(x, mainRecurCutoff=0, mainGrid=TRUE, mainXlabel=FALSE,
     data_frame <- waterfall_NA2gene(data_frame)
 
     # Plot the Heatmap
-    plot_x_title <- !is.null(clinData) && !plot_proportions
+    #plot_x_title <- !is.null(clinData) && !plot_proportions
+    plot_x_title <- ifelse(!is.null(clinData) || plot_proportions, FALSE, TRUE)
```
Comment (Alanocallaghan) [inline R/waterfall.R]: This line would be simpler as `plot_x_title <- !(!is.null(clinData) || plot_proportions)`

**Comment by Alanocallaghan** [comment]
Looks great thanks for spotting those edits

**Pull request merged**